google-cloud-datastore/src/test/java/com/google/cloud/datastore/BaseKeyTest.java

```diff
@@ -71,13 +86,29 @@ public void testBadDatasetInConstructor() throws Exception {
   @Test(expected = IllegalArgumentException.class)
   public void testBadDatasetInSetter() throws Exception {
     Builder builder = new Builder("d", "k");
+    builder.setProjectId(" ");
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testBadDatasetInSetterDeprecated() throws Exception {
+    Builder builder = new Builder("d", "k");
     builder.projectId(" ");
   }
 
   @Test
   public void testNamespace() throws Exception {
     Builder builder = new Builder("ds", "k");
     BaseKey key = builder.build();
+    assertTrue(key.getNamespace() != null);
+    assertTrue(key.getNamespace().isEmpty());
```

Comment:
Totally understood -- I ask questions sometimes just to understand the choices rather than as an objection.  I wouldn't give an LGTM if I was objecting or focused on something.